nemo/collections/cv/models/visual_ctc_bpe_models.py

```diff
@@ -0,0 +1,300 @@
+import copy
+import os
+from typing import Dict, List, Optional, Union
+
+import torch
+from omegaconf import DictConfig, ListConfig, OmegaConf, open_dict
+
+from nemo.collections.asr.losses.ctc import CTCLoss
+from nemo.collections.asr.metrics.wer_bpe import WERBPE, CTCBPEDecoding, CTCBPEDecodingConfig
+from nemo.collections.asr.parts.mixins import ASRBPEMixin
+from nemo.collections.cv.data import video_to_text_dataset
+from nemo.collections.cv.models.visual_ctc_models import VisualEncDecCTCModel
+from nemo.core.classes.common import PretrainedModelInfo
+from nemo.utils import logging, model_utils
+
+__all__ = ['VisualEncDecCTCModelBPE']
+
+
+class VisualEncDecCTCModelBPE(VisualEncDecCTCModel, ASRBPEMixin):
```

Comment:
Maybe you can add some comments in all of the new classes/functions on what exactly is different from the corresponding non-visual implementation. Will be easier to review and maintain in the future as there is a lot of duplication and probably only a few changes.